### PR TITLE
Uplink bucket attribution extension

### PR DESCRIPTION
**Pull request by aligeti**
What: 
Supports the Bucket's new attribution field

Why:
Value attribution functionality feature.

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

**Comment by jtolds** [comment]
how is this different than https://github.com/storj/storj/pull/2297 ? (which is merged)

**Comment by aligeti** [comment]
This adds the Attribution field to the stroj.Bucket and checks in OpenBucket () to see if that bucket needs to be attributed. Also the attribution for a bucket is stored in the DB.

**Comment by jtolds** [comment]
yes but you did that in the other PR also

**Comment by aligeti** [comment]
In the other PR, we just do the attribution, but never stored in the bucket’s meta info. This PR just extends what’s done on previous PR. Just an incremental change.

**Comment by jtolds** [comment]
okay this should be rebased against master

**Comment by aligeti** [comment]
Yes working on it.. I was working on this .. while master was being repaired.

File: lib/uplink/bucket_attrs_test.go [hunk 1]
```diff
@@ -72,28 +72,17 @@ func TestPartnerBucketAttrs(t *testing.T) {
 
 			partnerID := testrand.UUID().String()
 
-			consoleProjects, err := planet.Satellites[0].DB.Console().Projects().GetAll(ctx)
```
Comment (coyle) [inline lib/uplink/bucket_attrs_test.go]: still check the satellite db

Comment (aligeti) [reply lib/uplink/bucket_attrs_test.go]: updated

File: lib/uplink/bucket_attrs_test.go [hunk 2]
```diff
@@ -85,15 +85,17 @@ func TestPartnerBucketAttrs(t *testing.T) {
...
 			require.NoError(t, err)
 			assert.Equal(t, info.PartnerID.String(), partnerID)
 
-			// partner ID NOT set
+			// partner ID not set but bucket's attribution already set(from above)
 			proj.uplinkCfg.Volatile.PartnerID = ""
 			got, err = proj.OpenBucket(ctx, bucketName, &access)
 			require.NoError(t, err)
+			assert.Equal(t, got.bucket.Attribution, partnerID)
 			defer ctx.Check(got.Close)
```
Comment (ethanadams) [inline lib/uplink/bucket_attrs_test.go]: Should the defer be before assert.Equal?

Comment (aligeti) [reply lib/uplink/bucket_attrs_test.go]: moved the defer to before the assert

**Pull request merged**